lib/robots/roborock/capabilities/RoborockDoNotDisturbCapability.js

```diff
@@ -0,0 +1,58 @@
+const DoNotDisturbCapability = require("../../../core/capabilities/DoNotDisturbCapability");
+
+const DoNotDisturbAttribute = require("../../../entities/state/attributes/DoNotDisturbAttribute");
+
+class RoborockDoNotDisturbCapability extends DoNotDisturbCapability {
+    /**
+     * This function polls the current consumables state and stores the attributes in our robotState
+     *
+     * @abstract
+     * @returns {Promise<Array<import("../../../entities/state/attributes/DoNotDisturbAttribute")>>}
+     */
+    async getDnd() {
+        const data = await this.robot.sendCommand("get_dnd_timer", [], {});
+
+        const dnd = [
+            new DoNotDisturbAttribute({
+                enabled: (data[0].enabled === 1),
+                start: {
+                    hour: data[0].start_hour,
+                    minute: data[0].start_minute
+                },
+                end: {
+                    hour: data[0].end_hour,
+                    minute: data[0].end_minute
+                }
+            })];
+
+        dnd.forEach(d => this.robot.state.upsertFirstMatchingAttribute(d));
```

Comment:
DnD shouldn't be part of the RobotState but instead queried manually when it is required.

The RobotState should exclusively contain state information and no configurations